### DISCUSSION TITLE
Question about setting specific initial condition in MOOSE simulation

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#23454] [Q&A Modules: Phase field] [answered]
Hi everyone,

I still have not figured out how to set a specific states as the initial condition in MOOSE simulation.
I want to set the initial condition as the following shape, which is inside the red dotted line. How can I get the concentration and other variables I need from this file and put it in the new simulation? Can anyone give me some advice?
![image (6)](https://user-images.githubusercontent.com/63679626/219519017-bf9d4bdd-e244-4524-b244-c952017489aa.png)
I have tried SolutionUserObject. But it seems it does not work. Does anyone familiar with SolutionUserObject? Or have other methods?

Thank you very much,

**Comment by Rahim-Habibi** [comment, marked as answer]
Hi, 
If you have the initial state data as .csv, I can recommend to use "PropertyReadFile" then define it as a function. Last step, you can use the function as ICs.
